util/git/creds.go

```diff
@@ -1,18 +1,31 @@
 package git
 
 import (
+	"context"
+	"crypto/sha256"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"os"
 	"strings"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
 
 	argoio "github.com/argoproj/gitops-engine/pkg/utils/io"
+	"github.com/bradleyfalzon/ghinstallation"
 	log "github.com/sirupsen/logrus"
 
 	certutil "github.com/argoproj/argo-cd/util/cert"
 )
 
+// In memory cache for storing github APP api token credentials
+var githubAppTokenCache *gocache.Cache
+
+func init() {
+	githubAppTokenCache = gocache.New(60*time.Minute, 1*time.Minute)
```

Comment:
Please create `ARGOCD_GITHUB_APP_CREDS_EXPIRATION_DURATION` so we can override duration. This is a very low effort by might be a good workaround for someone in case of e.g. performance issues